_source/_posts/2023-07-28-scim-workshop.md

```diff
@@ -0,0 +1,1144 @@
+
+
+In this workshop, we'll cover the following:
+
+{% include toc.md %}
+
+In this workshop, you will learn how to implement SCIM and support provisioning from multiple organizations/tenants. Although we'll integrate with Okta as an example, remember that almost every identity provider on the web has SCIM support! 
+
+# Understanding SCIM 
+
+{% img blog/scim-workshop/scim-diagram.jpg alt:"Diagram of SCIM workflow" %}{: .center-image }
+
+## What problems does SCIM solve?
+There are many challenges with managing user identities across services:
+- SaaS developers want to know as soon as someone joins a company so they can set up an account 
+- SaaS developers also want to know when someone leaves a company to deactivate their account and associated resources.
+- Manually managing users through CSVs
+- Manually adding accounts at scale
+
+SCIM, the System for Cross-domain Identity Management, is an open standard which addresses those problems. SCIM: 
+- Provides near-instant updates to the downstream system whenever someone joins, moves inside of, or leaves a company
+- Allows management of user identities in cloud-based applications and services - i.e., you will have awareness of your users' up-to-date information
+- Enables automated control of user lifecycle (governance): provisioning (joiner), synchronizing user changes (movers), and de-provisioning (leaver) between identity providers (IdPs) and service providers (SPs)
+- Promotes interoperability via an open standard, which means allowing systems to exchange and make use of info being shared, e.g., I have worked with partners who have used SCIM as a solution to role-based access to resources 
+
+## Why do we love it?
+
+SCIM is a very well-designed standard; therefore, it can be implemented across systems that are compliant. It simplifies user identity management, thereby reducing manual admin tasks vs. having to manage user lifecycles across multiple applications. It allows user identity management by providing insight into up-to-date user information vs. having no governance over user information and assigned resources across services. Above all, it contributes to improved security! And now, who doesn't want that in this day and age?
+
+Now that I've provided you with some background on SCIM, let's move on to building the server!
+
+# Implement SCIM in a SaaS App 
+
+Before we build our SCIM server, let's have a plan of action. 
+
+- Learn about functions that support SCIM: Create, Read, Update, Delete Functions and JSON formatted requests and responses
+- Install the sample application and set up your development environment
+- Build the SCIM Interface in the Todo sample app and test it with Postman
+
```

Comment:
@edunham can you please move the scim image -  _source/_assets/img/blog/scim-workshop/scim-diagram.jpg - here instead. ty!